[Playground] Added check for IE11 for playground services

anishagg17 opened the pull request:
### Summary

`react-live` doesn't work for IE11 (react-view is based on react-live) thereby disabling it for IE11.

### Checklist

~- [ ] Check against **all themes** for compatibility in both light and dark modes~
- [ ] Checked in **mobile**
- [ ] Checked in **IE11** and **Firefox**
~- [ ] Props have proper **autodocs**~
~- [ ] Added **[documentation](https://github.com/elastic/eui/blob/master/wiki/documentation-guidelines.md)**~
~- [ ] Checked **[Code Sandbox](https://codesandbox.io/)** works for the any docs examples~
~- [ ] Added or updated **[jest tests](https://github.com/elastic/eui/blob/master/wiki/testing.md)**~
~- [ ] Checked for **breaking changes** and labeled appropriately~
~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~
~- [ ] A **[changelog](https://github.com/elastic/eui/blob/master/wiki/documentation-guidelines.md#changelog)** entry exists and is marked appropriately~

kibanamachine commented:
Since this is a community submitted pull request, a Jenkins build has not been kicked off automatically. Can an Elastic organization member please verify the contents of this patch and then kick off a build manually?

cchaos commented on src-docs/src/routes.js:
```diff
@@ -262,7 +262,18 @@ const createExample = (example, customTitle) => {
+    !!window.MSInputMethodContext &&
+    !!document.documentMode
+  ) {
+    console.warn(
+      "Playground service isn't available in IE11! Please use Edge or another modern browser."
+    );
+    playgroundComponent = null;
```
I would be a little more intrusive and actually make this a `window.alert` to warn the user and then only accept "OK" which then doesn't go to the route at all, but just keeps them on the original page.

anishagg17 replied:
can you run jenkins Build?

thompsongl commented:
jenkins test this

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_3757/

thompsongl commented:
Getting the same `bundle.js` error as before

anishagg17 commented:
> Getting the same `bundle.js` error as before

I checked it for several hours and came to know that there is some `object destructuring` which is not supported by `IE11`.
I am looking for a fix and would soon update the pr

anishagg17 commented:
@thompsongl I have fixed the earlier problem but still it isn't working because we need to `transpile` 10-13 more node_modules , these modules, because they use some syntax such as `class based` , `object destructuring` or `shorthand notation` for objects. I have fixed it a little bit but don't know how to fix it completely. Can you please look into this matter?

thompsongl commented:
> I have fixed the earlier problem

Which earlier problem?

I'll look into a way to transpile the lib

anishagg17 commented:
> Which earlier problem?

`object destructuring` by using `@babel/plugin-transform-destructuring`

thompsongl commented:
Update: I have it so IE11 can parse the bundle without errors, but the site still does not load. Unsure of the problem.

anishagg17 commented:
> Update: I have it so IE11 can parse the bundle without errors, but the site still does not load. Unsure of the problem.

Did you forget to return null from the `playgroundCreator` utility?
